Adds new input plugin for Azure Monitor

TooManyBees opened the pull request:
### Required for all PRs:

<!-- Complete the tasks in the following list. Change [ ] to [x] to
show completion. -->

- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.

<!-- Link to issues that describe the need for the change. Issues
should include context that will help reviewers understand why the
change is needed.

Make sure to link issues and using a keyword like "resolves #1234".
https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

resolves #

<!-- Finally, include a summary of the code change itself. This
description should tell reviewers how the issues were resolved.

example: Fixed an off by one error in counter variable in type FooBar.

example: Added an input plugin to gather yak shaving metrics using
golang library yaktech/shaver. -->

Adds an input plugin for Azure Monitor metrics. (This PR will eventually be trashed and repointed to a GitHub fork)

TooManyBees committed:
Adds new input plugin for Azure Monitor

TooManyBees commented on plugins/inputs/azure_monitor/azure_monitor.go:
```diff
@@ -0,0 +1,157 @@
+	for _, value := range monitorResponse.Value {
+		name := value.Name.Value
+		// FIXME: the format of the the data we're pushing is unspecified right now
+		fields[name] = value.TimeSeries
```
I am assuming that these fields can contain any struct that is serializable to JSON.

jcudit commented on plugins/inputs/azure_monitor/azure_monitor.go:
```diff
@@ -0,0 +1,157 @@
+func (a *AzureMonitor) Gather(acc telegraf.Accumulator) error {
+	resp, err := a.makeRequest()
+	if err != nil {
+		return err
```
Other inputs have leveraged [`acc.AddError`](https://github.com/influxdata/telegraf/search?q=acc.AddError) and returned `nil` from their `Gather()` functions.

May be some usability benefits that come out of this pattern 🤷🏾

TooManyBees committed:
Correct metrics endpoint

TooManyBees committed:
Fix plugin restart issue (probably a crash)

Even though `acc.AddFields` accepts a map[string]interface{}, storing
arbitrary structs in that map leads to the plugin crashing (possibly because
it doesn't know how to serialize it?).

This commit reduces the data sent to a single float64, which stabilizes the
plugin, but possibly loses data if more than 1 data point is returned in a
response. Immediately following this fix we should perform some data
munging to make sure that all data is collected.

TooManyBees committed:
Bucket data points by timestamp

To ensure that we send all data points in a response, reduce the response's
nested arrays of data to a map of fields keyed by RFC3339 timestamp, and then
invoke `telegraf.Accumulator.AddFields` for each timestamp.
